spirv_glsl.cpp

```diff
@@ -10937,8 +10938,14 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 
 		if (execution_scope == ScopeSubgroup || memory == ScopeSubgroup)
 		{
-			if (!options.vulkan_semantics)
-				SPIRV_CROSS_THROW("Can only use subgroup operations in Vulkan semantics.");
+			// OpControlBarrier with ScopeSubgroup is subgroupBarrier()
+			if (!options.vulkan_semantics && opcode != OpControlBarrier)
+			{
+				// Subgroup memory barriers are replaced by workgroup memory barriers if not vulkan_semantics
+				memory = ScopeWorkgroup;
+				execution_scope = ScopeWorkgroup;
```

Comment:
not sure what changing these two variables will do.

Reply:
workgroup mem barrier and subgroup mem barrier have same SPIRV opcode but differs in Memory field. You have to read code below

Reply:
yeah, but you can't emit a barrier in a non compute/non eval shader, so its best to just emit the statement you need and `break` like the case below for TES